### ISSUE TITLE
Large purge blob created by MMU2 when using single filament

### ISSUE THREAD

**Issue by rogerfroud** [prusa3d/Prusa-Firmware#2069]
When I print with a single colour using the MMU2, it always pauses at the front LH corner and outputs a large blob while stationary. It does this whether it loads having selected the job from the SD card, or if I 'Load to Extruder' beforehand.

This causes problems if there's a slightly solidified piece of filament stuck on the nose of the nozzle because it can't escape since there's no sideways motion.

![20190801_202211](https://user-images.githubusercontent.com/53141660/62429678-6e063500-b709-11e9-91ff-79bf7804cd27.jpg)

**Comment by michalxfanta** [comment]
Which version of firmware do you use?

**Comment by rogerfroud** [comment]
It's 3.7.2-2363

**Comment by tutulino** [comment]
Same here, even sometimes extruder start clicking.

**Comment by rambopierce** [comment]
Don’t know for sure, but I’m wondering if this isn’t some bad startup gcode from the slicer.  I do recall having a version of the slicer or left over startup gcode in the slicer configuration of a previous version that caused a similar issue for me.

**Comment by vintagepc** [comment]
I've observed this too, primarily with more "liquid" filaments (e.g. PLA at PETG temps from a material change) the incoming filament and trapped air can easily force out the slug of old filament. Often there'll be a snap/pop as the trapped air escapes and it'll stop oozing until the prime line.

**Comment by Daghis** [comment]
I'll add my +1 to this under the same circumstances, when printing something in MMU2S Single mode. It was bad enough that it ended up creating a clog between the PTFE tube and the hot end (Mosquito, in my case). 

Looking around, in `mmu_load_to_nozzle()`, it looks like it's extruding a whole bunch at one spot. Could this be responsible for the behavior we're seeing?

**Comment by mmcglumphy** [comment]
Also want to chime in to say that I've seen this multiple times. It can lead to jams when loading the first color, because it tries to push a large amount of filament out with the nozzle only slightly above the bed. Can lead to horrible grinding in the extruder. 

Edit:  FW 3.7.2, 1.0.6

**Comment by ACETyr** [comment]
This is not only happening in single mode. Also I've noticed it several times during a multi material print at times when no filament change is due. Printer happily prints and all of a sudden without any visible reason (no jams, no extruder clicking, no indication of any issue) it stops, moves the printhead to a different position and ejects the filament - but no cooling moves or other usual MMU2 stuff - just like when you use the "Eject Filament" command in the menu. Then you get the ".. replace old filament.." message on the lcd and the mmu spits out the filament for inspection.
After cutting the tip and recovering by pushing the button filament gets loaded, some gets extruded and you get the "Did color change correctly?" question on the lcd. If you answer Yes the printhead returns to the previous position over your print and extrudes a huge blob - ruining the print for good.

Happened 4 times today alone..

Firmware 3.8.1, MMU2 firmware 1.0.6. I've seen this behaviour in prior firmware versions also.

**Comment by rogerfroud** [comment]
Any progress on this from Prusa?

**Comment by vintagepc** [comment]
> This is not only happening in single mode. Also I've noticed it several times during a multi material print at times when no filament change is due. Printer happily prints and all of a sudden without any visible reason (no jams, no extruder clicking, no indication of any issue) it stops, moves the printhead to a different position and ejects the filament - but no cooling moves or other usual MMU2 stuff - just like when you use the "Eject Filament" command in the menu. Then you get the ".. replace old filament.." message on the lcd and the mmu spits out the filament for inspection.
> After cutting the tip and recovering by pushing the button filament gets loaded, some gets extruded and you get the "Did color change correctly?" question on the lcd. If you answer Yes the printhead returns to the previous position over your print and extrudes a huge blob - ruining the print for good.
> 
> Happened 4 times today alone..
> 
> Firmware 3.8.1, MMU2 firmware 1.0.6. I've seen this behaviour in prior firmware versions also.

Your issue sounds like false filament runouts. Calibrate/check your IR sensor.

**Comment by ACETyr** [comment]
I recalibrated the sensor after the first fail, I measured the sensor after the second fail I swapped the sensor after the 4th fail and I measured the sensor -> einsy cables while wiggling them - they checked out ok.

**Comment by leptun** [comment]
I think you need to check the FINDA, not the ir sensor. That is the one that regusters a runout.

**Comment by ACETyr** [comment]
Thanks leptun for pointing in that direction! It is the FINDA that's making problems in a way (but not the way I would have expected...) or at least the status the printer thinks the FINDA has.
I checked the FINDA and calibration seemed spot on. Also tripple checked the cables and did not find a fault. Started some small testprints and switched to Sensor Status on the LCD but problem did not appear. Then I made a print (rather big one) and on the 2nd layer on a very long diagonal move the issue happened again. I was able to glimpse FINDA status switching from 1 to 0 for a brief moment before the printer went into his unloading routine.
Out of curiosity I loaded the filament to nozzle and then lowered the finda until it nearly touched the ball so that I definately trigger (falsely) the FINDA to get a constant filament reading (even if no filament was present in truth). I reprinted the gcode and that time on a long diagonal fill move on layer 5 it happened again.
Either I have a faulty FINDA that refuses to show the error in my tests or for some reason the printer thinks FINDA status changes although it does not in reality.
The fact that the issue did not manifest in my small testprints but only when printing large objects might be pure coincidence - it baffles me and I have no explanation for that.
Other than ordering a new FINDA on my expense is there any test I can do to troubleshoot further?  

None the less: If the printer would refrain from producing a large blob after recovering from a perceived filament runout all prints would not have been lost. So there may be an issue with my FINDA but there is definitely an issue with how the printer behaves after a (perceived) filament runout...

**Comment by JohnOCFII** [comment]
I think @ACETyr issue is different than the OP.  I also have the same issue as the OP (ie, only during the starting purge line, I _think_ only on MMU single material prints), but my sensor display is showing the expected state, and not varying state when it shouldn't be.

**Comment by rogerfroud** [comment]
Could the administrator separate the two completely different issues that are being discussed here? The original thread is about a big blob forming at the start of every print, nothing else. That's what I'd like to see the answer to.

**Comment by ACETyr** [comment]
Didn't want to hijack the thread. When I initially chimed in I did not realize there was a different reason for the behaviour (although I think it is the same code being executed by the printer for other reasons). Apologies!

**Comment by rogerfroud** [comment]
Could someone look into this issue please? I've just downloaded the latest firmware for the MK3S 3.8.1 and mmu2 3.8.2 using PrusaSlicer 2.1.0 and I'm still getting this huge blob. Although that isn't a problem if it manages to extrude all of that material, sometimes it just can't because the head isn't moving. It really shouldn't be doing this.

It happens every time, so it ought to be very easy to replicate. I happens when you select to print from SD card, selecting the MMU2 channel, or if you load to nozzle first.

**Comment by psfbc** [comment]
Hello,

Same problem with my recent MMU2S build. Don't know if it is provoked by the T code or by the two lines Bellow:

G1 X55.0 E29.0 F1073.0
G1 X5.0 E29.0 F1800.0

But the quantity of material being extruded in the same spot is huge and has as consequence that the load fails. Maby the nozzle could be higher during the load?

Regards,
Paulo Carmo

**Comment by rogerfroud** [comment]
It's good to hear I'm not the only one with this problem, I get it on every single print with single colour MMU2 printing.

It would be great if someone from Prusa would pick this up and do something about it.

**Comment by brad-stone** [comment]
+1
I see the exact same problem when doing single filament prints - a really large blob in the left hand corner before the line prints across the front of the print bed.  As far as I can tell, it happens every single time; however, sometimes I hear the print head gears grinding as it tries to force out the filament quite rapidly.

i3 MK3s Firmware: 3.8.1-2869
MMU2s Firmware: 1.0.6-372
Octoprint: 1.3.12
PrusaSlicer: 2.1.0+win64

**Comment by rogerfroud** [comment]
Come on Prusa, we've shown this is easy to replicate, it's going to be an easy thing to fix so what's the hold up?

**Comment by vintagepc** [comment]
If it's so easy to fix, why not fix it yourself and submit a pull request? ;)

In all seriousness, they've been working on buttoning up the 3.9 release for the past little while now and all of the internal testing/fixes/tweaks that accompany that. They have an internal list of features/fixes planned out for whatever the pending release is (and often the one after that too) unless something show-stopping shows up, it's not going to get put in the release; it's simply not possible to manage a release with "fix ALL the things" and "add ALL the features. Yes, it sucks if your own issues are not on that list, but unfortunately that's simply the way software development is organized - someone had to make a judgement call at some point that leaves more than one person unhappy, and griping about it isn't going to get things fixed any faster. 

If you do want to take a stab at fixing it yourself, the relevant section is in Marlin_main.cpp; specifically the handling of Tx and Tc codes (The literals, not the T0-4). Since the blobbing doesn't happen on purge towers, the problem more or less has to be confined to that section and not in further shared code. Without diving in further I'm betting it's the split load-to-bondtech, wait, finish-load that's introducing some extra unexpected distance as opposed to the normal single-shot load during a multimaterial print.

Or, for a non-firmware workaround, you can likely modify your startup gcode to not use Tx/Tc and instead set the filament explicitly, with a T# call. You could make 5 "single mode" profiles, one for each slot.

**Comment by rogerfroud** [comment]
I don't think it's up to purchasers to fix bugs on an individual basis, that's for the system designer to do.

I'm well aware that they have lots of things to do, but if there's no response at all to the thread which is now months old, how are we to know they've even registered it as a bug?

I'm not about to spend ages fixing something that is not my problem. Not only is that a waste of my time, but are you seriously suggesting that everyone who has this problem has to also fix it themselves? All things have bugs, but this is pretty obvious to anyone who's used it in single colour mode. It seems that all of the testing has been done on multi-colour prints and none on single colour, else they would have picked this up before they even shipped it.

**Comment by vintagepc** [comment]
I'm just pointing out you have options if Prusa's internal timeframe for getting to it is unsatisfactory.

This is their pubic bug tracker, so yes, it has been registered for them to look at (along with the 702 other open issues...). Someone at Prusa has at least seen it (Michal Fanta is the product manager)

What I was suggesting is that there's nothing preventing you from fixing the bug and then sharing your fix here for possible inclusion in an upcoming release. This will undoubtedly get it moving a little faster, because now their only remaining work is to validate that it fixes the issue and doesn't cause any new ones; the legwork of identifying the cause and implementing a fix, by far the biggest task, is already done.

Look, I get you're unhappy about the state of affairs here and speed at which things are moving, that's pretty clear. I'm just trying to offer some perspective from the other side of the curtain on why every outstanding issue can't be addressed on the schedule its creator wants. From a software dev's perspective, the words "it'll be easy" and "while you're in there" are words of doom, things are rarely as simple as they seem on the surface. We hear those words and we know we're in for the long haul at the expense of something else...

**Comment by TheMrPhantom** [comment]
Having the same problem in mmu single mode

**Comment by rogerfroud** [comment]
If you update to all the latest firmware and Slicer, the problem in MMU2 seems to be much less ie the blob size is smaller than it used to be.

What frustrates me, is that if the software developer had once printed using MMU2 Single mode, he would have immediately seen that this was an issue. It just goes to show that they don't bother checking the software in the obvious ways that people will use it. Most people who have MMU2 won't be printing in multiple colours or with soluble supports, they'll be using MMU2 Single, so Prusa should at least do one trial print in that mode!

**Comment by scooper4711** [comment]
+1 for this issue. Firmware 3.9.0-3421. Prusa slicer 2.2.0

**Comment by mmcglumphy** [comment]
I've made this issue a little better for myself by adjusting the Start G-Code in Slic3r. I've changed it so that it purges the blob 1.5mm above the sheet instead of .4mm. Also, I made it so it doesn't retrace as far back as the blob on the return bead.

It could probably be tweaked further, but it's way better for now. Also, I'm not sure why the blob comes out, unless it has something to do with the Tc line where it selects the extruder.

Here are relevant lines that I changed at the bottom of the full Start G-Code...

;go outside print area
G1 Y-3.0 F1000.0
; start higher up at Z1.5 and not at Z0.4
G1 Z1.5 F1000.0
; select extruder
Tc
; purge line
G1 X55.0 F2000.0
G1 Z0.3 F1000.0
G92 E0.0
G1 X240.0 E25.0 F2200.0
G1 Y-2.0 F1000.0
; off set this section to the right by 10mm
G1 X65.0 E25 F1400.0
G1 Z0.20 F1000.0
; offset to the right 10mm to avoid blob
G1 X15.0 E4.0 F1000.0

M221 S{if layer_height<0.075}100{else}95{endif}
G92 E0.0

**Comment by FlashTech** [comment]
Thank you @mmcglumphy because I will be trying that. My issue is perfectly described by this issue's Title and I tried comparing the "MMU2S" vs "MMU2S single" Start G-code and all I can figure is that the command Tc extrudes too much compared to T[initial_tool]. I have had this problem throughout my many firmware and slicer upgrades. I literally bend over with a thin spatula and scrap the excess filament that sticks around the nozzle during the initial-purge-line to avoid small globs falling into the print later. I love that regular MMU2S mode does not have this problem at all.
![20200627_114733_HDR](https://user-images.githubusercontent.com/9146138/85928731-69d88e00-b86c-11ea-8d35-4bcb9035c5e5.jpg)


i3 MK3s Firmware: 3.9.0-3421
MMU2s Firmware: 1.0.6-372
PrusaSlicer: 2.2.0+win64
Printing from SD Card

**Comment by Photogad** [comment]
> 
> 
> I've made this issue a little better for myself by adjusting the Start G-Code in Slic3r. I've changed it so that it purges the blob 1.5mm above the sheet instead of .4mm. Also, I made it so it doesn't retrace as far back as the blob on the return bead.
> 
> It could probably be tweaked further, but it's way better for now. Also, I'm not sure why the blob comes out, unless it has something to do with the Tc line where it selects the extruder.
> 
> Here are relevant lines that I changed at the bottom of the full Start G-Code...
> 
> ;go outside print area
> G1 Y-3.0 F1000.0
> ; start higher up at Z1.5 and not at Z0.4
> G1 Z1.5 F1000.0
> ; select extruder
> Tc
> ; purge line
> G1 X55.0 F2000.0
> G1 Z0.3 F1000.0
> G92 E0.0
> G1 X240.0 E25.0 F2200.0
> G1 Y-2.0 F1000.0
> ; off set this section to the right by 10mm
> G1 X65.0 E25 F1400.0
> G1 Z0.20 F1000.0
> ; offset to the right 10mm to avoid blob
> G1 X15.0 E4.0 F1000.0
> 
> M221 S{if layer_height<0.075}100{else}95{endif}
> G92 E0.0

thank you

**Comment by Jeltesteur** [comment]
+1 for this occuring to a recent mmu2s upgrade to prusa mk3s with firmware 3.9.1-3518
+1 thank you to mmcglumphy for the work around
Jelte

**Comment by FEsmondeWhite** [comment]
This is the same experience I just had with a newly installed MMU2, latest firmware on mk3s, with latest PrusaSlicer RC and default filament profile.

This giant blob in single-material print mode is not the normal exquisite Prusa experience!

I paused my print to clean the nozzle immediately during the purge line. I don't understand how this could make it through testing...

**Comment by rogerfroud** [comment]
Frankly I don't think this has been tested else it would have been picked up. My guess is that all of the testing done with the MMU2 is done with more than one filament and a wipe tower. I suspect the issue doesn't happen when you do that, but I may be wrong.
However, I use my MMU2 with a single filament without a wipe tower all off the time, and it's about time someone at Prusa stepped up and sorted this out!
Does anyone at Prusa read these threads?

**Comment by vintagepc** [comment]
I'm thinking this isn't really a firmware issue (the prime line is generated by the slicer), I think it would be better filed over on the Prusaslicer issue list where it could be more easily addressed.

**Comment by FEsmondeWhite** [comment]
I posted to twitter as well... I'll cross post to the PrusaSlicer issue list. I'm surprised no one at Prusa seems to have replied to this.

As far as I can tell, I don't have any mechanical/electrical issues in my new setup. This shouldn't be the out of box experience!

**Comment by vintagepc** [comment]
In all fairness I have some insight into what the goings on are... it just boils down to this sort of thing just being much lower on the priority list. It's annoying when it happens, but the printer still works fine - as opposed to things like LA1.5 not working right, or the superPinda/other hardware supplier support. 

Not to mention they're basically on a skeleton crew in house in the midst of a pandemic.

**Comment by rtyr** [comment]
This is firmware issue. It happens right after `Tc` command (before purge line gcode generated by PrusaSlicer).

**Comment by FEsmondeWhite** [comment]
@wavexx @DRracer

A giant blob is caused when starting prints in single filament sliced prints using mk3s + MMU2.

Have you seen this issue?

**Comment by FEsmondeWhite** [comment]
The Prusa devs are *amazing*. The recent work on firmware and PrusaSlicer has been exceptional. Between new features and fixes, the Prusa team continues to deliver!

Please also consider that this bug occurs on every single mode print, can break printers, and should have a fairly simple fix. It's also really easy to test since it occurs right after heating & probing.

> In all fairness I have some insight into what the goings on are... it just boils down to this sort of thing just being much lower on the priority list. It's annoying when it happens, but the printer still works fine - as opposed to things like LA1.5 not working right, or the superPinda/other hardware supplier support.
> 
> Not to mention they're basically on a skeleton crew in house in the midst of a pandemic.

**Comment by Area5142** [comment]
I print primarily in PETG and this firmware error results in strings being pulled and debris getting stuck on the nozzle, even on the coated non-stick ones.
I try to clean the nozzle before each print, but this error prevents a neat print even before it starts.
Maybe raising the nozzle before the _tc_ g-code will help, as suggested by @mmcglumphy.

**Comment by wavexx** [comment]
@FEsmondeWhite I don't have an MMU unit :), but I do remember we fixed a problem earlier on this year that would cause the first extrusion to jerk in the first move immediately after a G92. The move would be worse when it was combined with the Z move (lowering to the build plate).

I wonder if this could be a similar situation?

Could somebody post a complete serial log when this happens?

**Comment by vintagepc** [comment]
The nozzle doesn't lower to the build plate in this case, it's already parked at the correct Z height at the beginning of the prime line after the home.

**Comment by wavexx** [comment]
@rtyr do you have a minimal repro for this already? (just up to the first G1 of the purge)

**Comment by FEsmondeWhite** [comment]
> 
> 
> @rtyr do you have a minimal repro for this already? (just up to the first G1 of the purge)

I can run testing for you.

> Could somebody post a complete serial log when this happens?

What would you like me to log? Just plug the einsy into usb and log the serial output from the printer during the start of printing?

I can send you the associated gcode as well.

**Comment by wavexx** [comment]
@FEsmondeWhite having the gcode (and serial log too would help) cut up to the point where it happens would be a good start.
If you feel especially willing, removing all the gcode parts that are not relevant to the issue would help even more (so that even if I don't have experience with the mmu I can still have a look)

**Comment by rtyr** [comment]
@wavexx just look at Slack, it is described there.

**Comment by rtyr** [comment]
It is about `Tc` command. Its purpose is to load the filament further to the nozzle before the print. It apparently loads too much filament for "S" extruder (shorter path than non-S) and creates a blob. It usually doesn't cause any issues, but it is not optimal. The blob on the first photo looks too big though (probably incorrect IR sensor calibration).

**Comment by rogerfroud** [comment]
I don't see how it can be incorrect IR sensor calibration when you get great prints. It's just sitting there stationary at the start of every print, extruding for no good reason for several seconds. This ought to be so easy to fix. Why would it ever extrude when it's stationary and at the level of the first layer?

**Comment by rtyr** [comment]
@rogerfroud If it is extruding for several seconds, than you have some different issue (most likely IR sensor related).

**Comment by FEsmondeWhite** [comment]
Hi @rtyr @wavexx

Here are some test results... I have (bad) videos, but think video is probably not very helpful. Serial output from Putty is in notes.txt.

Tested two start configurations:
1. Filament loaded to MMU only.
2. Filament loaded to MMU and at nozzle.

Note that the printer crashes & reboots when starting a single-mode print with filament already loaded to the nozzle.

![IMG_20201230_121155446_HDR](https://user-images.githubusercontent.com/35807628/103369564-772f3900-4a98-11eb-95c8-8bc6b7b0c9cd.jpg)
[Tc-test-gcode.txt](https://github.com/prusa3d/Prusa-Firmware/files/5755028/Tc-test-gcode.txt)
[notes.txt](https://github.com/prusa3d/Prusa-Firmware/files/5755030/notes.txt)

**Comment by rtyr** [comment]
> 
> 
> Why would it ever extrude when it's stationary and at the level of the first layer?

If you look at the start g-code you will see Tx and Tc commands. Tx command asks user which filament to use and loads the filament, but only to the Bondtech gears. Then printer waits for temps, homes and do MBL......then Tc command loads the filament further to the nozzle.

**Comment by FEsmondeWhite** [comment]
After bed probe, the extruder moves to the bed surface at the start of the purge line.

It stays in place at the front-left side of the bed for about 11 seconds while starting the purge. The extruder pulls back, then pushes forward, at two different speeds. This period is when the large blob is formed. Then the extruder starts moving afterward and makes the forward/backward purge lines.

**Comment by rtyr** [comment]
> 
> 
> After bed probe, the extruder moves to the bed surface at the start of the purge line.
> 
> It stays in place at the front-left side of the bed for about 11 seconds while starting the purge. The extruder pulls back, then pushes forward, at two different speeds. This period is when the large blob is formed. Then the extruder starts moving afterward and makes the forward/backward purge lines.

Correct, but it doesn't extrude for several seconds like rogerfroud suggested. Easy way to see what is happening is to preheat the printer, send Tx command, select filament+wait for load, send Tc command.

**Comment by FEsmondeWhite** [comment]
It does extrude for a while. I agree that the prime distance is incorrectly configured for the mk3s. I'm assuming this shouldn't be a difficult fix, considering the mk3s without MMU does this perfectly.

I am linking a short clip from the video so you can see how long it sits there for. I don't have a spinner on my extruder motor, and it's hard to see when the E-axis motor is spinning.

https://photos.app.goo.gl/9FQvdadiRQALqATT7

**Comment by vintagepc** [comment]
^^ If it was an IR sensor issue then T# [0-5] would also be affected (that's what non-single mode MMU slices use). Is there a reason we do the split load while cold for single mode slices rather than just wait until just before prime like a "real" MMU print?

So somehow, filament push distance from  Tx + Tc != T0..4

Looking at the code - @FEsmondeWhite, do you also get a blob if you remove Tx and Tc, and replace them with a single `T?` **AFTER the nozzle has heated?** (Right before the first G1... E###. 

For example:
```
G1 Y-3.0 F1000.0
G1 Z0.4 F1000.0
; select extruder
T?
; initial load
G1 X55.0 E32.0 F1073.0
```

This would (I think) make the load more like a full MMU slice in a single shot and narrow down whether this is specific to Tx/Tc handling or in common loading code like mmu_load_to_nozzle

**Comment by vintagepc** [comment]
.. May have found it - Assuming these comments are correct, this is the cause - mmu.cpp:790 - mmu_load_to_nozzle();

//! This function is used only only after T? (user select filament) and M600 (change filament).
//! It is not used after T0 .. T4 command (select filament), in such case, gcode is responsible for loading
//! filament to nozzle.
//!

If I recall, the MMU M600, will extrude extra filament after the load in order to prime and purge - since there is no allocated space on the wipe tower and it may potentially need to resume in the middle of a printed part. That extra purge is likely the cause of our blob during the prime line. 

And if we adjust that function to not produce the blob, it will likely cause gaps in the user's print if they are executing an M600 manually or due to runout. 

Tc/? and M600s are obviously outside the domain of the slicer - the firmware cannot know the correct distance to continue loading in this scenario and so has hard-coded values - further complicating the matter as it will not be correct for anything other than the extruder revision for which it was coded.

**Comment by FEsmondeWhite** [comment]
Yes, there is still a small blob at the start of the print.

Here is the modified gcode:
[Tc-test-2-gcode.txt](https://github.com/prusa3d/Prusa-Firmware/files/5755135/Tc-test-2-gcode.txt)

Video of test:
https://photos.app.goo.gl/PeVTWGay3kakEfAh6

**Comment by vintagepc** [comment]
Thank you. That confirms my analysis of the above in regards to mmu_load_to_nozzle as this function is called both during Tc and T?

Seems the least invasive way to deal with this would potentially be to follow the T# model - have the slicer generate the gcode for the initial Tc load from gears to nozzle and using the various "Single extruder MM" parameters as appropriate (like it does for T0..4) rather than relying on Tc to do it for us during the purge line. What do you think, @rtyr ?

**Comment by FEsmondeWhite** [comment]
> Thank you. That confirms my analysis of the above in regards to mmu_load_to_nozzle as this function is called both during Tc and T?

My pleasure!

That makes sense. It's obviously complicated to try to track the printer state across a wide variety of conditions with so many potential states in the MMU.

If you need additional testing support, please let me know.

It would be *very* helpful if the mk3s+MMU2 could be operated in MK3S-only mode, where the user would be required to load the filament to the nozzle before prints like with a mk3s. That would reduce corner cases to be handled in firmware, and reduce the complexity for user printing.

**Comment by FEsmondeWhite** [comment]
> Seems the least invasive way to deal with this would potentially be to follow the T# model - have the slicer generate the gcode for the initial Tc load from gears to nozzle and using the various "Single extruder MM" parameters as appropriate (like it does for T0..4) rather than relying on Tc to do it for us during the purge line.

I don't know much about the Tx/Tc/T?/T# commands, as I just started working with the MMU. That does sound like a good plan.

Thank you again for looking at this so quickly!

**Comment by vintagepc** [comment]
The Tc/T? are obviously intended as a convenience shortcut for that step. Personally though when I'm doing a lot of single material/colour stuff I'll just bypass the MMU and feed straight to the extruder. (I usually do a lot of the same colour and so just want to avoid the hassle of selecting/reloading/having it unload at the end of the print) But note I'm also running a custom mod of the firmware that uses both sensors during MMU printing - else you'd lose runout detection since you need to stick something in the FINDA to keep it sensing filament. (PR #2499 )

**Comment by mmcglumphy** [comment]
I mostly print in single material mode, as well. I have the MMU for when I need more than one material. I also like the auto spool join feature for burning though old spools of gray. Though because of this blob issue I use the multi profile in Slic3r, so I don't have the blob. But then I lose the ability to start any prints from other than the first slot.

Hehe. My luck that this issue is finally being look at after all this time. I finally grew weary of constantly swapping materials out of slot one and just removed my MMU altogether this past week. Overall, I find it way too fiddly and wasteful and not up to the same standards as the MK3.

**Comment by rtyr** [comment]
> 
> Seems the least invasive way to deal with this would potentially be to follow the T# model - have the slicer generate the gcode for the initial Tc load from gears to nozzle and using the various "Single extruder MM" parameters as appropriate (like it does for T0..4) rather than relying on Tc to do it for us during the purge line. What do you think, @rtyr ?

I will leave it to the firmware guys. I don't think it will be so easy, there are also some checks with Tc command (filament goes forward/backward) and slicer doesn't know if filament is already loaded to the nozzle or not (not typical but possible scenario). I think the correct way is to adjust Tc command for MK3S/2.5S in the FW.

**Comment by FEsmondeWhite** [comment]
Note that if filament is already loaded to the nozzle, it causes my printer to restart(!).

Thanks again for looking into this issue.

I looked at trying to use multimaterial slicing and just use one filament spool, but it wouldn't slice because I'm running a print where different objects have some altered layer heights. It refused to slice because this would cause problems with a wipe tower. Though obviously I don't need a wipe tower with single color printing...

**Comment by rogerfroud** [comment]
> After bed probe, the extruder moves to the bed surface at the start of the purge line.
> 
> It stays in place at the front-left side of the bed for about 11 seconds while starting the purge. The extruder pulls back, then pushes forward, at two different speeds. This period is when the large blob is formed. Then the extruder starts moving afterward and makes the forward/backward purge lines.


> @rogerfroud If it is extruding for several seconds, than you have some different issue (most likely IR sensor related).

Since I created the thread, and everyone is showing the same issue, I fail to see how I can have an IR sensor issue while others with the same issue don't. That doesn't make sense.

 It's worked like this from day one of installing the MMU2. The printer was working just fine before the installation, and it's produced superb prints since. If there was an issue with the IR sensor I can't see how it would do that.

The big blob is the only issue. It just sits there for a while with the head down and extruding before moving off. Why would it ever extrude when not moving? That's what needs to be investigated. There's no need to ever do that as far as I can see.

**Comment by FEsmondeWhite** [comment]
I guess it's worth emphasizing that in MMU multi-color mode, printing with just one color, this blob doesn't happen at the start of prints. I spent some more time in PrusaSlicer, and disabled the wipe tower in multi-color mode to be able to use multicolor mode (non-Single mode). I think this should work for my single-color printing needs. 
![image](https://user-images.githubusercontent.com/35807628/103381007-7b6b4e80-4ab8-11eb-9208-a95478b6ce51.png)

The start code uses T4 to set the extruder filament (I'm selecting filament 5), and it works more or less like a standard mk3s, without the blob. Maybe PrusaSlicer could simply use this approach to set the filament selection and prime the extruder? This would fix the issue before the firmware team can dig into it. I guess the idea is to have the user manually select the filament at print-start time, hence using "Tx/Tc/T?". But if T4 works without the blob, maybe there's a way to fix "T?" priming without significant changes...

```
;go outside print area
G1 Y-3.0 F1000.0
M73 Q0 S62
G1 Z0.4 F1000.0
; select extruder
T4
; initial load
G1 X55.0 E29.0 F1073.0
M73 P0 R61
G1 X5.0 E29.0 F1800.0
G1 X55.0 E8.0 F2000.0
G1 Z0.3 F1000.0
G92 E0.0
G1 X240.0 E25.0 F2200.0
G1 Y-2.0 F1000.0
G1 X55.0 E25 F1400.0
G1 Z0.20 F1000.0
G1 X5.0 E4.0 F1000.0
G92 E0.0
```

@rogerfroud I agree it's not the IR sensor. I think that at this point, that was not the suspected issue by anyone.

**Comment by rtyr** [comment]
@rogerfroud
I believe I already explained how and why the overextrusion/blob happens.

I mentioned the IR sensor only as possible reason why is the blob on the 1st picture bigger and as possible reason for "extruding for several seconds on one spot" you mentioned..

**Comment by FEsmondeWhite** [comment]
> > After bed probe, the extruder moves to the bed surface at the start of the purge line.
> > It stays in place at the front-left side of the bed for about 11 seconds while starting the purge. The extruder pulls back, then pushes forward, at two different speeds. This period is when the large blob is formed. Then the extruder starts moving afterward and makes the forward/backward purge lines.
> 
> Correct, but it doesn't extrude for several seconds like rogerfroud suggested. Easy way to see what is happening is to preheat the printer, send Tx command, select filament+wait for load, send Tc command.

@rtyr: @rogerfroud is correct, it does extrude for several seconds while stationary (the initial Tx/Tc purge lasts around 11 seconds). It only does this in single material mode, not using multimaterial slicing and a single filament. Please see the associated videos for confirmation/evidence.

While IR probe false triggering will cause random retraction/feed issues, that's not the issue causing the purge blob here. In my testing the IR sensor is properly calibrated, and the print runs without errors for hours after the initial purge blob is cleared

**Comment by rtyr** [comment]
> @rtyr: @rogerfroud is correct, it does extrude for several seconds while stationary (the initial Tx/Tc purge lasts around 11 seconds). It only does this in single material mode, not using multimaterial slicing and a single filament. Please see the associated videos for confirmation/evidence.
> 
You can clearly see on your video, that it is extruding under 1 second (0:34-0:35) ;-).

**Comment by rtyr** [comment]
0:34-0:35 that is extruding :-)

**Comment by rtyr** [comment]
Anyway, I guess it is clear now, that he is experiencing the same "standard" behavior as on your video. I will leave this issue to FW team.

Happy New Year

**Comment by FEsmondeWhite** [comment]
Happy new year!

Thanks for getting eyes on this issue,

Francis

**Comment by wjbuys** [comment]
I'm seeing exactly the same issue (extruding a big blob at the start of the purge line) in MMU Single mode, latest firmware on both the MMU and MK3S+, latest PrusaSlicer and configuration profiles.

With this behaviour, it's infuriating to try and print with PETG since bits of the blob stick to the nozzle and end up ruining the first layer. It also chews up the filament in the bondtech gears, since it can't actually push more plastic into the blob. This causes underextrusions later on in the skirt where the thinner filament makes it into the nozzle.

**Comment by Area5142** [comment]
I see the same big blob of filament in MMU2 Single mode. To avoid the PETG sticking too much to the nozzle, I changed the startup g-code to raise the nozzle to 4 mm for the first part of the introduction line.

**Comment by rogerfroud** [comment]
Come on Prusa, please do something about this, I raised this issue in August 2019!!!

**Comment by FEsmondeWhite** [comment]
It is very frustrating that prusa haven't resolved this issue.

The workaround for me has been to use the slicer in multifilament mode for
single filament prints, then disable the wipe tower and just use filament 1
for all single filament prints.

This also means I need to change the filament #1 spool between jobs,
instead of being able to set the spool # for each job on the fly
(pre-sliced parts).

Francis

**Comment by rogerfroud** [comment]
There are other issues with the MMU2 firmware that they need to address if
you're only using it for single material prints. I'm trying to put together
a scenario they can repeat. I'm finding that if it fails to unload the
filament successfully at the end of the print, say, it then gets stuck in a
logical loop that you can't get out of where it thinks the MMU2 has an
issue. Turning it off and on again, or trying to force an unload is to no
avail. It can take ten minutes of messing about to finally clear the error
so it can start working normally.
Reset doesn't help. If anyone has a similar problem and knows what triggers
it and how to get out, I'd be interested to know.

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by FEsmondeWhite** [comment]
When it gets stuck during unloading (maybe 15% of the time), it's usually
from a blob or string on the end of the filament at the mmu in the Bowden
time. I unscrew the Bowden from the selector, and cut off the filament tip.
Usually by holding down the middle button on the mmu2, it will then re-try
the retraction, and complete the unloading. Sometimes I reset the mmu2 with
the button by the mmu2 usb port.

In my case, I installed a slightly longer Bowden tube because the normal
Bowden was slightly kinked with the stock length, and during loading it
runs a sequence of slightly longer pushes until it finally reaches the
extruder gears.

A similar issue to the unloading is when it has trouble loading at some
point during a print (often at the start, with a fresh cut tip). Many mmu2
lights start blinking. After addressing the issue, it gets stuck in the
same loop where the printer won't recognize that the physical problem is
resolved, and the mmu2/mk3s are not in sync. The mmu2 is correctly loaded,
but the mk3s is still paused waiting on the mmu2. Usually forcing the mk3s
to heat back up, then resetting the mmu2 will make it start again.

Francis

**Comment by mari01337** [comment]
Same blob problem here in single mode...

**Comment by rogerfroud** [comment]
There doesn't seem to be any ongoing development of the MMU2 as far as I can tell, no new version of the firmware for over a year. Come on Prusa, can we get some bug fixes on this!
The firmware is pretty flaky in my opinion once it runs into trouble. You can't clear any fault that results in the MMU2 flashing its lights. Nothing seems to clear the issue, certainly turning it off, or pressing Reset doesn't work. The LCD display talks about pressing a button on the MMU2, but it doesn't say which one, and I've tried pressing all of them and it's as if they aren't connected to anything.
It can get into a state where it just won't load, or the wheels keep spinning as if it's trying to load the filament, but nothing is happening. It also goes as it it's going to select the right filament again, but then just ends up with the lights on the MMU2 flashing.
This really is pretty hopeless as it is, I'm forever getting stuck with being unable to clear the fault and get the filament loaded again. Please try to use this and create faults, prevent it from unloading by releasing the grip on the extruder, it's easy enough to do.
Please also see why it keeps failing to unload, this happens frequently. At the moment, the MMU2 is a complete pain in the neck, I wish I'd never bought it.

**Comment by vintagepc** [comment]
FYI it was announced in one of the live-streams that they are reworking the MMU FW  - which is probably why the existing MMU FW hasn't seen much movement. 

https://youtu.be/HgwusaEWNuk?t=4362 (around 1h:12m)

**Comment by rogerfroud** [comment]
> FYI it was announced in one of the live-streams that they are reworking the MMU FW - which is probably why the existing MMU FW hasn't seen much movement.
> 
> https://youtu.be/HgwusaEWNuk?t=4362 (around 1h:12m)

Let's hope so, because it's hopeless at the moment. It seems like they got it so that it almost worked and then walked away. It's very frustrating when people don't test products thoroughly. If anyone at Prusa had used this in Single mode for just a few days, they would have found all of these problems.

**Comment by FEsmondeWhite** [comment]
I am not sure if this was ever improved through the FW or PrusaSlicer, but the easy workaround is to slice single filament prints using the MK3S printer profile, instead of the MK3S+MMU2 printer profile. The printer complains about the gcode being sliced for the wrong printer when starting a print, but prints properly like a non-MMU MK3. This also allows the filament to be manually loaded once, and it won't retract at the end of every print. The downside is that you don't get the spool join feature.

**Comment by KoDaKrom** [comment]
Same issue, I would be appreciated if PRUSA could do something to resolve this blob caused by the Tc command.
Actual mitigation in the Start G-code is to lift higher a little bit the Z coordinate avoiding clicking extruder and/or put the nozzle in the blob.
But a clean firmware update would be a professionnal resolving way.
Thank you in advance Pursa team ;-)

EDIT: Could you please also take in count the new E3D Revo Six hotend internal geometry ?!?

**Comment by Prusa-Support** [comment]
Thank you all for your input.

As already emerged in this conversation, the main difference between the printer profiles "MK3x (standalone printer)" and "MK3x MMU2 Single (single filament)" is that the latter profile causes the printer to push filament into the nozzle further because the filament is not supposed to be in the nozzle: the filament should be withdrawn by the Multi Material module at the end of a print, and "re-loaded to nozzle" at the beginning of the next print.

If there is filament in the hotend before the start of an "MMU2 single" print, the printer will push extra filament (it should be for a length of 57mm) creating a blob at the beginning of the starter purge line.
Basically, at the moment the printer doesn't take into account the possibility that filament is already "loaded to nozzle", so please make sure the filament was correctly withdrawn back to the initial position in the MMU2 module before the print.

For those of you who intentionally want to avoid unloading filament at the end of the "MM single print", disconnecting the MMU2 could be a viable way to walk around the unload-reload behavior.
Please notice this is not considered a standard scenario: the MMU2 needs to be ready to start a new print from a different extruder/position once the print is completed.

The developers already considered changing this behavior, but, unfortunately, we don't have official news about such an update yet. I can't tell if this behavior will change in the next MM-control board firmware release. 
The MMU2 firmware has not been updated for a while because a large portion of it is being re-written, supposedly for higher efficiency and for simpler interaction with the user via LCD (besides the LEDs combination).

For now, I'd recommend leaving the issue open as a request for loading filament process enhancement.
Thanks for your feedback.

Michele Moramarco
Prusa Research

**Comment by scooper4711** [comment]
Please note - it's not so much that we want to avoid unloading filament at the end of the MM Single Print, it's more that getting the MMU mod to *load* filament is such a crapshoot that we tend to leverage "load to nozzle" to get the filament to load. It only takes about 5-10 attempts, but it's better to do so before the print starts so you can make good use of the reset button when it fails to load and you have to disassemble something.

**Comment by Prusa-Support** [comment]
@scooper4711 I'm no sure if we are talking about an enhancement of the existing feature anymore.
If I understand correctly, the printer is erratically failing to load filament in your case. If this is the case, some troubleshooting is required to identify the problem and possibly sort it out once and for all.

Suggestions to improve or ease the loading before a "MM single filament print" are very appreciated. The issue will remain open to that.

If we are looking at a different problem, or an anomaly that is not yet a confirmed firmware problem, please feel free to open a new separate issue, or possibly reach our Customer Support for troubleshooting.
https://help.prusa3d.com/article/customer-support_2287
Feel free to write an email to my attention (Michele Moramarco), with reference to this issue, so that we can clarify the nature of the described problem together.

Michele Moramarco
Prusa Research

**Comment by gudnimg** [comment]
I suspect the issues with `Tx` and `Tc` gcodes are fixed with 3.13. It at least no longer happens when starting a single color print with no filament loaded prior.

**Comment by Prusa-Support** [comment]
Please test FW Einsy 3.13.0 + MMU 3.0.0 .
The MMU firmware has been totally rewritten for improved user experience and more reliability.

Blobs on the purge tower shouldn't be expected.

Regarding issues related to the long loading distance in MM-single mode instead, the problems wouldn't persist as long as the filament is unloaded after the print, which is the only scenario the printer will correctly work with.
We acknowledge some users strongly want to skip the un/loading operations in MM-single mode, but based on tests, we can't let that happen - and sadly we even had to close the PR https://github.com/prusa3d/Prusa-Firmware/pull/4212 about that.

The new firmware release and the act of closing a relevant PR renders this issue obsolete.

Michele Moramarco
Prusa Research

**Comment by rogerfroud** [comment]
I'm afraid you're years too late, I'm done with it. I reported the
horrendous and frustrating issues with  errors that can't be cleared,
wasting hours trying to get the single filament use to work, and there was
zero response and support.
I've gone back to loading manually because all I wanted was to use the MMU
for soluble supports. I can live without that functionality to avoid the
pain of dealing with an MMU.
If you don't support your customers, they will turn their back on you.
Roger

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by FEsmondeWhite** [comment]
I also removed my MMUs, they weren't reliable and broke the basic mk3 user
experience. I don't trust new Prusa gear anymore. I used to evangelize for
Prusa, but won't make the same mistake with a mk4 or XL. I'm not interested
in investing days of my time or hundreds of dollars into upgrading to
become an unsupported Prusa test engineer again, so I probably won't try an
MMU 3. I have a friend with a Bambu, maybe I'll ask him to do occasional
multicolor prints for me instead?

I understand closing this issue, since the solution by Prusa was to never
fix the issues on the MMU 2. The issue apparently doesn't exist on the MMU
3.

Francis

**Comment by Prusa-Support** [comment]
We appreciate your sincere opinion.
Your feedback was a great help in improving the firmware and the product.

I encourage you to reach out the Customer Support.
Aided by the new firmware, we are able to run troubleshooting more accurately than ever. 
Every day we find a suitable solution in pretty much all reported cases.
All we need is the customer's collaboration.
https://help.prusa3d.com/article/customer-support_2287

Speaking about this issue, as far as I can see, it can be closed and fixed now.
Nonetheless, if you have relevant feedback about this issue, we would gladly listen to you.

Michele Moramarco
Prusa Research
